maxim/logger/components/generation.py

```
@@ -200,12 +226,24 @@ def __init__(
             self.provider = "unknown"
         self.messages.extend([m for m in (final_config.get("messages") or [])])
+        self.messages, attachments = parse_attachments_from_messages(self.messages)
+        if len(attachments) > 0:
+            for attachment in attachments:
+                self.add_attachment(attachment)
         self.model_parameters = parse_model_parameters(
             final_config.get("model_parameters", {})
         )
 
     @staticmethod
     def set_provider_(writer: LogWriter, id: str, provider: str):
```

Ruff (0.11.9)

Missing return type annotation for staticmethod `set_provider_`

Add return type annotation: `None`

(ANN205)